### PR TITLE
Golang: Improve error handling

### PR DESCRIPTION
**Pull request by davinci26**
While working on the golang filter I saw the following improvements we can make
in the error handling:

1. Expose stats for panics and generally adding `stats.h` for the golang filter
2. Surface panic messages to envoy error logs

This is achieved by moving the implementation of panic error handling from `SendLocalReply` to
a dedicated function that can add the o11y for the errors before we surface them to the users.

Also removed `Golang` from the response message when the golang filter panics to avoid leaking implementation
details to end users.

Signed-off-by: Sotiris Nanopoulos <sotirisnan@gmail.com>

### CODE REVIEW

**Comment by davinci26** [comment]
Need to figure out how to compile the fuzz code, because it doesnt repro locally. It must be that my OS skips this code

File: examples/golang/envoy.yaml [hunk 1]
```diff
@@ -17,7 +22,7 @@ static_resources:
             typed_config:
               "@type": type.googleapis.com/envoy.extensions.filters.http.golang.v3alpha.Config
               library_id: simple
-              library_path: "lib/simple.so"
+              library_path: "bazel-bin/examples/golang/simple/simple.so"
```
Comment (doujiang24) [inline examples/golang/envoy.yaml:25]: is this only for local test?

**Review by doujiang24 — COMMENTED** [review]
Nice, thank you very much.
one small comment, otherwise, LGTM.

File: examples/golang/envoy.yaml (hunk 1 above)
Comment (davinci26) [reply examples/golang/envoy.yaml]: yeah, reverted the change in that file

File: contrib/golang/filters/http/source/cgo.cc [hunk 2]
```diff
@@ -169,6 +169,14 @@ void envoyGoFilterHttpFinalize(void* r, int reason) {
   delete req;
 }
 
+CAPIStatus envoyGoFilterHttpSendPanicReply(void* r, void* details) {
+  return envoyGoFilterHandlerWrapper(r, [details](std::shared_ptr<Filter>& filter) -> CAPIStatus {
+    // Deep clone the GoString into C++, since the GoString may be freed after the function
+    // returns, while they may still be used in the callback.
+    return filter->sendPanicReply(copyGoString(details));
```
[the remark this answers is not in the document]
Comment (davinci26) [reply contrib/golang/filters/http/source/cgo.cc:176]: yeah avoiding the copy here makes sense

**Review by doujiang24 — APPROVED** [review]
LGTM, thanks.

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**